### PR TITLE
feature: Add coverage with api token :breaking:

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ $ bash <(curl -Ls https://coverage.codacy.com/get.sh) report --help
 Command: report
 Usage: codacy-coverage-reporter report 
   --project-token | -t  <your project API token>
+  --api-token | -t  <your project API token>
```
Comment (prcr) [inline README.md:33]:
```suggestion
  --api-token | -t  <your account API token>
```

File: src/main/scala/com/codacy/rules/ReportRules.scala [hunk 2]
```diff
@@ -79,8 +81,8 @@ class ReportRules(coverageServices: => CoverageServices) extends LogSupport {
       if (config.partial) {
         logger.info(
           """To complete the reporting process, call coverage-reporter with the final flag.
-          | Check https://docs.codacy.com/coverage-reporter/adding-coverage-to-your-repository/#multiple-reports
-          | for more information.""".stripMargin
+            | Check https://docs.codacy.com/coverage-reporter/adding-coverage-to-your-repository/#multiple-reports
```
Comment (prcr) [inline src/main/scala/com/codacy/rules/ReportRules.scala:84]: I just noticed this link is outdated (although it includes a redirect to the new one):

```suggestion
            | Check https://docs.codacy.com/coverage-reporter/#multiple-reports
```

**Comment by prcr** [comment]
After you fix the CI it will hit another issue until we merge this fix https://github.com/codacy/codacy-coverage-reporter/pull/311 and rebase.

File: src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala [hunk 3]
```diff
@@ -75,11 +75,13 @@ case class Report(
 case class BaseCommandConfig(
     @Name("t") @ValueDescription("your project API token")
     projectToken: Option[String],
-    @Name("a") @ValueDescription("your api token") @Hidden
+    @Name("a") @ValueDescription("your account api token")
     apiToken: Option[String],
-    @Name("u") @ValueDescription("your username") @Hidden
+    @Name("o") @ValueDescription("organization provider")
+    organizationProvider: Option[OrganizationProvider.Value],
+    @Name("u") @ValueDescription("your username")
```
Comment (josemiguelmelo) [inline src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala:82]: Can't `username` be a bit confusing for the user as it can be used both for organization repositories and for the user account? Wouldn't it be better to call something like organization or owner name (or at least on the description) to be in the same line as the `organizationProvider`? 🤔

File: src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala [hunk 4]
```diff
@@ -75,11 +75,13 @@ case class Report(
 case class BaseCommandConfig(
     @Name("t") @ValueDescription("your project API token")
     projectToken: Option[String],
-    @Name("a") @ValueDescription("your api token") @Hidden
+    @Name("a") @ValueDescription("your account api token")
     apiToken: Option[String],
-    @Name("u") @ValueDescription("your username") @Hidden
+    @Name("o") @ValueDescription("organization provider")
```
Comment (lolgab) [inline src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala:80]: Do you need to define a "single char" way to pass this parameter?
I understand that `p` is already taken so you went for `o`, but I wouldn't define any and leave only the long option as we have for other parameters.

File: src/main/scala/com/codacy/model/configuration/Configuration.scala [hunk 5]
```diff
@@ -1,7 +1,8 @@
 package com.codacy.model.configuration
 
-import java.io.File
+import com.codacy.api.OrganizationProvider
 
+import java.io.File
```
Comment (lolgab) [inline src/main/scala/com/codacy/model/configuration/Configuration.scala:5]: The order of the imports changed here. Can you revert the `java.io.File` import where it was and put the `com.codacy.api.OrganizationProvider` import in alphabetical order together with the other codacy imports?

File: src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala [hunk 6]
```diff
@@ -1,7 +1,8 @@
 package com.codacy
 
-import java.io.File
+import com.codacy.api.OrganizationProvider
 
+import java.io.File
```
Comment (lolgab) [inline src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala:5]: The order of the imports changed here. Can you revert the `java.io.File` import where it was and put the `com.codacy.api.OrganizationProvider` import in alphabetical order together with the other codacy imports?

File: src/main/scala/com/codacy/rules/ConfigurationRules.scala [hunk 7]
```diff
@@ -139,7 +141,7 @@ class ConfigurationRules(cmdConfig: CommandConfiguration, envVars: Map[String, S
         "Maybe you forgot the http:// or https:// ?"
       }
       Left(s"""$error
-              |$help""".stripMargin)
+           |$help""".stripMargin)
```
Comment (lolgab) [inline src/main/scala/com/codacy/rules/ConfigurationRules.scala:144]: Can you revert this change?

File: src/main/scala/com/codacy/rules/ConfigurationRules.scala [hunk 8]
```diff
@@ -149,6 +151,7 @@ class ConfigurationRules(cmdConfig: CommandConfiguration, envVars: Map[String, S
     *
     * This function try to get the API base URL from environment variables, and if not
     * found, fallback to the public API base URL
+    *
```
Comment (lolgab) [inline src/main/scala/com/codacy/rules/ConfigurationRules.scala:154]: Is this Intellij adding these extra lines?

File: src/main/scala/com/codacy/rules/ReportRules.scala [hunk 9]
```diff
@@ -51,7 +51,9 @@ class ReportRules(coverageServices: => CoverageServices) extends LogSupport {
           _ <- storeReport(report, file)
           language <- guessReportLanguage(finalConfig.languageOpt, report)
           success <- sendReport(report, language, finalConfig, commitUUID, file)
-        } yield { success }
+        } yield {
+          success
+        }
```
Comment (lolgab) [inline src/main/scala/com/codacy/rules/ReportRules.scala:56]: Unrelated change.

File: src/main/scala/com/codacy/rules/ReportRules.scala [hunk 10]
```diff
@@ -77,11 +79,9 @@ class ReportRules(coverageServices: => CoverageServices) extends LogSupport {
         }
       }
       if (config.partial) {
-        logger.info(
-          """To complete the reporting process, call coverage-reporter with the final flag.
-          | Check https://docs.codacy.com/coverage-reporter/adding-coverage-to-your-repository/#multiple-reports
-          | for more information.""".stripMargin
-        )
+        logger.info("""To complete the reporting process, call coverage-reporter with the final flag.
+            | Check https://docs.codacy.com/coverage-reporter/#multiple-reports
```
Comment (lolgab) [inline src/main/scala/com/codacy/rules/ReportRules.scala:83]: Nice that you fixed the link here 👌

**Review by lolgab — CHANGES_REQUESTED** [review]
Some little suggestions here and there but it looks pretty cool!
So we removed the api-token method from the docs some time ago because of the missing provider that made it impossible to work correctly?

File: README.md [hunk 11]
```diff
@@ -30,6 +30,10 @@ $ bash <(curl -Ls https://coverage.codacy.com/get.sh) report --help
 Command: report
 Usage: codacy-coverage-reporter report 
   --project-token | -t  <your project API token>
+  --api-token | -t  <your account API token>
+  --organization-provider | -o  <the project provider> (manual, gh, bb, ghe, bbe, gl, gle)
```
Comment (jllopes) [inline README.md:34]: Let's keep it consistent 😄 
```suggestion
  --organization-provider | -o  <the organization provider> (manual, gh, bb, ghe, bbe, gl, gle)
```

File: src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala (hunk 3 above)
Comment (lolgab) [reply src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala:82]: Consider that `username` was already there even if `@Hidden` annotated. So changing it means breaking the CIs of users. Right?

File: src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala (hunk 6 above)
Comment (rtfpessoa) [reply src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala:5]: I did not do anything by hand. it was probably intellij. I do not think we should keep this kind of stuff ordered by hand.

File: src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala (hunk 3 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/configuration/parser/ConfigurationParser.scala:82]: yup, I would prefer not to change it.

File: src/main/scala/com/codacy/model/configuration/Configuration.scala (hunk 5 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/model/configuration/Configuration.scala:5]: Same as above

File: src/main/scala/com/codacy/rules/ConfigurationRules.scala (hunk 7 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/rules/ConfigurationRules.scala:144]: This was automatic, I did not do it by hand

File: src/main/scala/com/codacy/rules/ConfigurationRules.scala (hunk 8 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/rules/ConfigurationRules.scala:154]: yes

File: src/main/scala/com/codacy/rules/ReportRules.scala (hunk 9 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/rules/ReportRules.scala:56]: formatter

File: src/main/scala/com/codacy/rules/ReportRules.scala (hunk 10 above)
Comment (rtfpessoa) [reply src/main/scala/com/codacy/rules/ReportRules.scala:83]: it was @prcr

File: src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala (hunk 6 above)
Comment (lolgab) [reply src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala]: I tried to run the formatter on Intellij and it gave me a different result.
Anyway, it wasn't ordered by hand before but by the Metals "Organize imports" task.
It is ( I think ) a good practice to not include unrelated changes when doing a PR, but if you really want to change the imports order, please run the Optimize Import task on Intellij to fix this.

**Review by lolgab — APPROVED** [review]

File: src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala (hunk 6 above)
Comment (rtfpessoa) [reply src/it/scala/com/codacy/CodacyCoverageReporterSpec.scala]: It is shitty that we cannot have this on the terminal. Metals being different from intellij and different from scalafmt

**Comment by mrfyda** [comment]
@rtfpessoa after you merge this, let's revert #256 to get the docs back
